### PR TITLE
Bump repository to Go 1.17 and k8s v0.23.x 

### PR DESCRIPTION
**Pull request by timflannagan**
**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [x] Implementation matches the proposed design, or proposal is updated to match implementation
- [x] Sufficient unit test coverage 
- [x] Sufficient end-to-end test coverage
- [x] Docs updated or added to `/docs` 
- [x] Commit messages sensible and descriptive

### CODE REVIEW

File: Dockerfile.okd [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-FROM registry.ci.openshift.org/openshift/release:golang-1.16 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
```
Comment (timflannagan) [inline Dockerfile.okd:2]: Note: it looks like this is needed as the okd-related checks were failing with the following message:

```
[1/2] STEP 5/5: RUN make osbs-build
./build/build.sh
building marketplace-operator...
# sigs.k8s.io/json/internal/golang/encoding/json
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
```

**Review by dinhxuanvu — COMMENTED** [review]
/lgtm

**Pull request merged**